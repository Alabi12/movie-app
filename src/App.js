import React from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import Home from './components/Home/Home';
import MovieDetail from './components/MovieDetail/MovieDetail';
import './App.scss';

function App() {
  return (
    <div className="app">
      <Router>
        <div className="container">
          <Switch>
            <Route path="/movie/:imdbID" component={MovieDetail} />
            <Route path="/" component={Home} />
          </Switch>
        </div>
      </Router>
    </div>
  );
}

export default App;
