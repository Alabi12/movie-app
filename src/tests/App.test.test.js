import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { MemoryRouter as Router } from 'react-router-dom';
import store from '../redux/store';
import App from '../App';

describe('Testing App component', () => {
  test('Snapshot testing', () => {
    const myRenderer = renderer.create(
      <Provider store={store}>
        <Router>
          <App />
        </Router>
        ,
      </Provider>,
    );
    const testComponent = myRenderer.toJSON();
    expect(testComponent).toMatchSnapshot();
  });
});
