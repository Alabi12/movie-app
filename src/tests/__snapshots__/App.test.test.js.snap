// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing App component Snapshot testing 1`] = `
Array [
  <div
    className="app"
  >
    <div
      className="container"
    >
      <div>
        <div
          className="header"
        >
          <div
            className="search-bar"
          >
            <form
              onSubmit={[Function]}
            >
              <input
                onChange={[Function]}
                placeholder="search for movies"
                type="text"
                value=""
              />
              <button
                type="submit"
              >
                <i
                  className="fa fa-search"
                />
              </button>
            </form>
          </div>
        </div>
        <div
          className="banner-img"
        />
        <div
          className="movie-wrapper"
        >
          <div
            className="movie-list"
          >
            <div
              className="movie-container"
            >
              <div
                className="movies-error"
              >
                <h3 />
              </div>
            </div>
          </div>
          <div
            className="show-list"
          >
            <div
              className="movie-container"
            >
              <div
                className="shows-error"
              >
                <h3 />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  ",",
]
`;
